opendse-model/src/main/java/net/sf/opendse/model/TGFFSpecification.java

```diff
@@ -0,0 +1,151 @@
+package net.sf.opendse.model;
+
+import java.util.Set;
+
+import net.sf.opendse.model.parameter.Parameter;
+
+/**
+ * The {@code TGFFSpecification} consists of an {@link Application},
+ * {@link ResourceTypes}, and {@link Mappings}.
+ * 
+ * @author Valentina Richthammer
+ * 
+ */
+public class TGFFSpecification implements IAttributes {
+
+	protected ResourceTypes<?> resourceTypes = null;
+	protected Application<?, ?> application = null;
+	protected Mappings<?, ?> mappings = null;
+	protected Link wiring = null;
+	protected Attributes attributes = new Attributes();
+
+	/**
+	 * Constructs the specification. 
+	 * 
+	 * @param application
+	 *            the application
+	 * @param resourceTypes
+	 *            the set of resource types
+	 * @param mappings
+	 *            the type mappings
+	 */
+	public TGFFSpecification(Application<?, ?> application, ResourceTypes<?> resourceTypes, Mappings<?, ?> mappings, Link wiring) {
+		super();
+		this.resourceTypes = resourceTypes;
+		this.application = application;
+		this.mappings = mappings;
+		this.wiring = wiring;
+	}
+
+	/**
+	 * Returns the set of resource types.
+	 * 
+	 * @param <A>
+	 *            the type of architecture
+	 * @return the architecture
+	 */
+	@SuppressWarnings("unchecked")
+	public <L extends Link> L getWiring() {
+		return (L) wiring;
+	}
+	
+	/**
+	 * Returns the set of resource types.
+	 * 
+	 * @param <A>
+	 *            the type of architecture
+	 * @return the architecture
+	 */
+	@SuppressWarnings("unchecked")
+	public <R extends ResourceTypes<Resource>> R getResourceTypes() {
+		return (R) resourceTypes;
+	}
+
+	/**
+	 * Returns the application.
+	 * 
+	 * @param <A>
+	 *            the type of application
+	 * @return the application
+	 */
+	@SuppressWarnings("unchecked")
+	public <A extends Application<Task, Dependency>> A getApplication() {
```

Comment:
Could be defined in an interface "ApplicationProvider" with a single method: `public <A extends Application<Task, Dependency>> A getApplication()`